pkg/manager/impl/execution_manager.go

```diff
@@ -367,6 +367,43 @@ func (m *ExecutionManager) setCompiledTaskDefaults(ctx context.Context, task *co
 	checkTaskRequestsLessThanLimits(ctx, task.Template.Id, task.Template.GetContainer().Resources)
 }
 
+// Fetches inherited execution metadata including the parent node execution db model id and the source execution model id
+// as well as sets request spec metadata with the inherited principal and adjusted nesting data.
+func (m *ExecutionManager) getInheritedExecMetadata(ctx context.Context, requestSpec *admin.ExecutionSpec,
+	workflowExecutionID *core.WorkflowExecutionIdentifier) (parentNodeExecutionID uint, sourceExecutionID uint, err error) {
+	if requestSpec.Metadata == nil || requestSpec.Metadata.ParentNodeExecution == nil {
+		return parentNodeExecutionID, sourceExecutionID, nil
+	}
+	parentNodeExecutionModel, err := util.GetNodeExecutionModel(ctx, m.db, requestSpec.Metadata.ParentNodeExecution)
```

Comment:
Does this query the DB? if so, is that needed? don't we already have the node execution id in `requestSpec.Metadata.ParentNodeExecution`

Reply:
yes, because we need the model id to map the relationship

Reply:
ah because we need the DB id (int?) to store?

Reply:
yeah! s/model/DB/g